icu4c/source/test/intltest/rbbitst.cpp

```diff
@@ -4883,6 +4897,104 @@ void RBBITest::TestDebugRules() {
 #endif
 }
 
+void RBBITest::testTrieStateTable(int32_t numChar, UCPTrieValueWidth expectedTrieWidth,
+                                  int32_t expectedStateRowBits) {
+    // Text are duplicate characters from U+4E00 to U+4FFFF
+    UnicodeString text("");
+    for (UChar c = 0x4e00; c < 0x5000; c++) {
+        text.append(c).append(c);
+    }
+    // Generate rule which will caused length+4 character classes and
+    // length+3 states
+    UnicodeString rules("!!quoted_literals_only;");
+    for (UChar c = 0x4e00; c < 0x4e00 + numChar; c++) {
+        rules.append('\'').append(c).append(c).append("';");
+    }
+    rules.append(".;");
+    UErrorCode status = U_ZERO_ERROR;
+    UParseError parseError;
+    LocalPointer<RuleBasedBreakIterator> bi(new RuleBasedBreakIterator(rules, parseError, status));
+
+    assertEquals("Number of char classes mismatch", numChar + 4, bi->fData->fHeader->fCatCount);
+    assertEquals("Number of states mismatch", numChar + 3, bi->fData->fForwardTable->fNumStates);
```

Comment:
I think that relying on the precise number of states (or character classes) may end up being fragile, in the event that we make tweaks to the table generation code. Which is somewhat plausible.

Reply:
any suggestion what should we do here?

Reply:
ok to keep as is